yoda.sh

```diff
@@ -7,23 +7,69 @@ requires:
   - "Python:(?!osx)"
   - "Python-modules:(?!osx)"
   - "Python-system:(osx.*)"
+  - ROOT
 build_requires:
   - "autotools:(slc6|slc7)"
 prepend_path:
-  PYTHONPATH: $YODA_ROOT/lib/python3.9/site-packages
+  PYTHONPATH: $YODA_ROOT/lib/python3.9/site-packages # _Question 1_ : 3.9 hardcoded here ?
```

Comment:
See xrootd recipe https://github.com/alisw/alidist/blob/master/xrootd.sh#L84-L92
here just use a simple `python` and later (after the compilation and install), detect the actual python directory and create a `python` symlink to it.

Reply:
Hello Adrian, 
thanks for the pointer and the advice.
I have implemented the changes in the spirit of what is done for XRootD (a simplified version here).
2 questions:

1. Any issue to anticipate with this solution on the front of the relocation issues (moving from build machine to execution ones) ?
2. Any opinion on some alternative prepand_path command below, making use of the yoda-config command ?
```
   > echo `exec yoda-config --pythonpath`
/scratch/AliceSuite/sw/ubuntu2004_x86-64/YODA/v1.9.5-local4/lib/python3.9/site-packages
   > echo $(exec yoda-config --pythonpath) | gawk -F '/' '{print $(NF-2)"/"$(NF-1)"/"$(NF)}'
lib/python3.9/site-packages
```
i.e. in the recipe, sthg like:
```
prepend_path:
  PYTHONPATH: "$YODA_ROOT/`echo $(exec yoda-config --pythonpath) | gawk -F '/' '{print $(NF-2)"/"$(NF-1)"/"$(NF)}'`"
   #  Getting the path towards python site-packages from YODA, e.g. $YODA_ROOT/lib/python3.9/site-packages
```
or simpler, but likely creating issues with relocation (?) :
```
  PYTHONPATH: `echo `exec yoda-config --pythonpath`
```


In other words, I am wondering:

1. whether it is possible to invoke bash command within the instruction 
2. and what is immediately discoverable on-the-fly after the build via some commands (is yoda-config already accessible to be launched ?).

Reply:
@maireiphc Hi! Sorry, i did not seen your reply! so:
1. you do not want absolute paths (as not to have relocation problems)
you just need a `python` --> `python3.9` symlink (where `3.9` you get from the PYTHON_VER, see https://github.com/alisw/alidist/blob/master/xrootd.sh#L24)
so, https://github.com/alisw/alidist/blob/master/xrootd.sh#L95-L100 is all you need

2. prepend_path: dictionary whose key-value pairs are an environment variable name and a path to be prepended to it, as it happens in LD_LIBRARY_PATH. This happens only after the package declaring the prepend_path in question is built, so it is not available in the same recipe (just like variables declared using env).
from http://alisw.github.io/alibuild/reference.html#the-header

so, IMHO i would not use (or rely) on shell commands run in prepend_path

Basically, the problem is: given that in a similar context, the XROOTD recipe works for us, what is the problem with doing the same for yoda?

Reply:
Hello @adriansev, 
thanks for the reply and consideration on question 1 of the initial commit.
I consider the item resolved (what I implemented with commit https://github.com/alisw/alidist/pull/4473/commits/bfd895195bbf3a00d1b668e4493075525ed19998, second one here in this PR thread, does the work via symlink).
Do you confirm ?

Reply:
@adriansev : ping for a potential confirmation on this go-ahead (symlink towards python/site-package)

Reply:
@maireiphc yes, i confirm that this is the way to go (it's the only way to have a static PYTHONPATH declaration when the python bindings are local to the package)

Reply:
Great. Thanks for the feedback ! - I will declare now this "conversation as resolved".